game/src/lib.rs

```diff
@@ -0,0 +1,216 @@
+        g
+    });
+}
+
+fn smoke_test() {
+    let mut timer = Timer::new("run a smoke-test for all maps");
+    for name in abstutil::list_all_objects(abstutil::path_all_maps()) {
+        let map = map_model::Map::new(abstutil::path_map(&name), &mut timer);
+        let scenario = if map.get_city_name() == "seattle" {
+            abstutil::read_binary(abstutil::path_scenario(&name, "weekday"), &mut timer)
+        } else {
+            let mut rng = sim::SimFlags::for_test("smoke_test").make_rng();
+            sim::ScenarioGenerator::proletariat_robot(&map, &mut rng, &mut timer)
+        };
+
+        let mut opts = sim::SimOptions::new("smoke_test");
+        opts.alerts = sim::AlertHandler::Silence;
+        let mut sim = sim::Sim::new(&map, opts, &mut timer);
+        // Bit of an abuse of this, but just need to fix the rng seed.
+        let mut rng = sim::SimFlags::for_test("smoke_test").make_rng();
+        scenario.instantiate(&mut sim, &map, &mut rng, &mut timer);
+        sim.timed_step(&map, Duration::hours(1), &mut None, &mut timer);
+
+        if vec![
+            "downtown",
+            "krakow_center",
+            "lakeslice",
+            "montlake",
+            "udistrict",
+        ]
+        .contains(&name.as_str())
+        {
+            dump_route_goldenfile(&map).unwrap();
+        }
+    }
+}
+
+fn dump_route_goldenfile(map: &map_model::Map) -> Result<(), std::io::Error> {
+    use std::fs::File;
+    use std::io::Write;
+
+    let path = abstutil::path(format!("route_goldenfiles/{}.txt", map.get_name()));
+    let mut f = File::create(path)?;
+    for br in map.all_bus_routes() {
+        writeln!(
+            f,
+            "{} from {} to {:?}",
+            br.osm_rel_id, br.start, br.end_border
+        )?;
+        for bs in &br.stops {
+            let bs = map.get_bs(*bs);
+            writeln!(
+                f,
+                "  {}: {} driving, {} sidewalk",
+                bs.name, bs.driving_pos, bs.sidewalk_pos
+            )?;
+        }
+    }
+    Ok(())
+}
+
+fn check_proposals() {
+    let mut timer = Timer::new("check all proposals");
+    for name in abstutil::list_all_objects(abstutil::path("system/proposals")) {
+        match abstutil::maybe_read_json::<map_model::PermanentMapEdits>(
+            abstutil::path(format!("system/proposals/{}.json", name)),
+            &mut timer,
+        ) {
+            Ok(perma) => {
+                let map = map_model::Map::new(abstutil::path_map(&perma.map_name), &mut timer);
+                if let Err(err) = map_model::PermanentMapEdits::from_permanent(perma, &map) {
+                    timer.error(format!("{} is out-of-date: {}", name, err));
+                }
+            }
+            Err(err) => {
+                timer.error(format!("{} JSON is broken: {}", name, err));
+            }
+        }
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+use wasm_bindgen::prelude::*;
```

Comment:
I'm assuming that other than these wasm32 sections this file was cut/paste from the old game/src/main.rs

Reply:
Yep. git didn't detect the rename properly. :(